lib/src/client.rs

```diff
@@ -413,8 +410,41 @@ impl ClientInner {
                         return Err(Error::Consensus(BlockchainError(err)));
                     }
                 };
+                BlockchainProxy::from(&blockchain)
+            }
+            SyncMode::Light => BlockchainProxy::from(&Arc::new(RwLock::new(LightBlockchain::new(
+                config.network_id,
+            )))),
+        };
+
+        // Create the Dht verifier
+        #[cfg(feature = "full-consensus")]
+        let dht_verifier = Verifier::new(blockchain_proxy.clone());
+
+        // Create the network.
+        let network = Arc::new(
+            Network::new(
+                network_config,
+                #[cfg(feature = "full-consensus")]
+                dht_verifier,
```

Comment:
Even if we have the `full-consensus` feature enabled at compile time, we might still start a light client.